### PR TITLE
Exercise 1

### PR DESCRIPTION
**Pull request by aba-yom-mee**
Module 1 project exercise

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,6 +34,7 @@ share/python-wheels/
 *.egg-info/
 .installed.cfg
 *.egg
+.idea/
```
Comment (Jonesey13) [inline .gitignore:37]: Ideally you should be using vscode (so make sure your project exercise can run in vscode going forward)

File: app.py [hunk 2]
```diff
@@ -1,12 +1,41 @@
...
+
+    return redirect("/")
+
+@app.route('/delete/<id>', methods=['POST'])
+def delete_item(id):
+    item = session.get_item(id)
+
+    session.delete_item(item)
+
+    return render_template('index.html')
```
Comment (Jonesey13) [inline app.py]: If you're showing the homepage again you should probably use a redirect to ensure the top level navigation (i.e. the search bar) has the correct url.

File: feature.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+This is my new feature.
```
Comment (Jonesey13) [inline feature.txt]: You can probably delete this file

File: session_items.py [hunk 4]
```diff
@@ -67,3 +67,11 @@ def save_item(item):
     session['items'] = updated_items
 
     return item
+
+def delete_item(item):
+
+    existing_items = get_items()
+    existing_items.remove(item)
+    updated_items = existing_items
```
Comment (Jonesey13) [inline session_items.py]: This line is redundant (just use existing_items)

File: templates/index.html [hunk 5]
```diff
@@ -1,18 +1,66 @@
...
+<div class="well text-container">
+    <div class="row">
+        <div class="col-md-12">
+            <h2 style="color: blue">Items</h2>
+            <ul class="list-group mb-4">
+                <h6>
+                    {% for todo in todos %}
+                    {{ todo.title }} {{todo.status}}
+                    <a href="/" class="btn btn-danger">Item to Delete</a>
+                    <br></br>
```
Comment (Jonesey13) [inline templates/index.html]:
```suggestion
                    <br>
```

File: templates/index.html [hunk 6]
```diff
@@ -1,18 +1,66 @@
...
+    <p style="color: blue" class="lead">Just another to-do app.</p>
+</div>
+<div class="well text-container">
+    <div class="row">
+        <div class="col-md-12">
+            <h2 style="color: blue">Items</h2>
+            <ul class="list-group mb-4">
+                <h6>
+                    {% for todo in todos %}
+                    {{ todo.title }} {{todo.status}}
```
Comment (Jonesey13) [inline templates/index.html:16]: I'd put the content of the for loop one level of indentation in, e.g.
```
                    {% for todo in todos %}
                        {{ todo.title }} {{todo.status}}
                        <a href="/" class="btn btn-danger">Item to Delete</a>
                        <br></br>
                    {% endfor %}
```

File: templates/index.html [hunk 7]
```diff
@@ -1,18 +1,66 @@
...
+                    <a href="/" class="btn btn-danger">Item to Delete</a>
```
Comment (Jonesey13) [inline templates/index.html:17]: You'll want to link this to a delete endpoint (and probably move this to a form)

File: templates/index.html [hunk 8]
```diff
@@ -1,18 +1,66 @@
...
+            
+            <div class="col-md-12"></div>
+            <form action="/" method="POST">
+                    <h2>
+                        <label style="color: blue">Add New Item</label>
+                        <input class="form-control" id="name"  type="text" name="title">
+                        <button type="submit" class="btn btn-success">Create New Item</button>
+                        <br>
+                    </h2>
+                </div>
```
Comment (Jonesey13) [inline templates/index.html:37]:
```suggestion

```

File: templates/index.html [hunk 9]
```diff
@@ -1,18 +1,66 @@
...
+
+
+
+        <div>
+            
+            <div class="col-md-12"></div>
+            <form action="/" method="POST">
+                    <h2>
+                        <label style="color: blue">Add New Item</label>
+                        <input class="form-control" id="name"  type="text" name="title">
```
Comment (Jonesey13) [inline templates/index.html:33]: maybe add a `required` attribute to this to prevent empty todos

File: templates/index.html [hunk 10]
```diff
@@ -1,18 +1,66 @@
...
+        <div class="col-md-12">
+        <div class="well text-container">
+            <div class="row">
+                <form action="/save/{{Item}}" method="POST">
+                    <div class="form-group">
+                        <h2>
+                            <label style="color: blue">Title</label>
+                            <input class="form-control" type="text" name="Title">
+                            <button type="submit" class="btn btn-success">Save Item</button>
+                            <a href=" " class="btn btn-success">Save Item</a>
```
Comment (Jonesey13) [inline templates/index.html:50]: Link just navigates to the homepage

File: app.py [hunk 11]
```diff
@@ -1,12 +1,41 @@
...
+
+@app.route('/', methods=['POST'])
+def add_item():
+    session.add_item(request.form.get("title"))
+    return redirect("/")
+
+
+@app.route('/save', methods=['POST'])
+def save_item(item):
+    item = session.save_item(item)
```
Comment (Jonesey13) [inline app.py:26]: Item will need to read off the form data. However given you already have working functionality for this in the method above I'd just remove the entire method and the corresponding section in index.html

File: app.py [hunk 12]
```diff
@@ -1,12 +1,41 @@
...
 
+
 @app.route('/')
-def index():
-    return 'Hello World!'
+def get_items():
+    return render_template('index.html', todos=session.get_items())
+
+
+@app.route('/item/<id>')
```
Comment (Jonesey13) [inline app.py:13]: Route undiscoverable

File: templates/delete.html [hunk 13]
```diff
@@ -0,0 +1,29 @@
+{% extends "layout.html" %}
```
Comment (Jonesey13) [inline templates/delete.html:1]: template unused

**Review by Jonesey13 — APPROVED** [review]
Looks functional to me!

Left a few comments but feel free to merge this PR in when ready 👍

File: .gitignore (hunk 1 above)
Comment (aba-yom-mee) [reply .gitignore:37]: OK

**Pull request merged**